algorand/go-algorand
# docker: Do not catchup if already initialized.

## Summary

Avoid running fast catchup in docker containers which have already been initialized.

## Test Plan

1. Build docker container with these changes:
```
docker build \
           -t wwinder/algod:fast-catchup-fix  \
           --build-arg TARGETARCH=amd64 --no-cache \
           --build-arg BRANCH=will/docker-catchup-init \
           --build-arg URL=https://github.com/winder/go-algorand \
           .
```

2. Run a container such that it initialize a local data directory
```
docker run --rm -it -p 8080:8080 --name algod-test-run \
           -e ADMIN_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
           -e TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
           -e PROFILE=development \
           -e NETWORK=betanet \
           -e FAST_CATCHUP=1 \
           -v (pwd)/docker_fast:/algod/data \
           wwinder/algod:fast-catchup-fix
```

3. Observe that fast catchup runs, and completes:
```
curl "localhost:8080/v2/status?pretty" -H "Authorization: bearer aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```

4. Shutdown the container, and start it back up again with the same command from step 2.
5. Monitor status with command from step 3, it should not run fast-catchup.
6. Shutdown the container, start it back up again with `algorand/algod:3.18.0-stable`.
7. Monitor status with command from step 3, it should start fast-catchup (the bug).
